phys2denoise.py

```diff
@@ -0,0 +1,234 @@
+
+    Parameters
+    ----------
+    metric : function
+        Metric function that is being called
+    args : dict
+        Dictionary containing all arguments that are used to parametrise metric
+
+    Notes
+    -----
+    Outcome
+        An info-level message for the logger.
+    """
+    msg = f'The {metric} regressor will be computed using the following parameters:'
+
+    for arg in args:
+        msg = f'{msg}\n    {arg} = {args[arg]}'
+
+    msg = f'{msg}\n'
+
+    LGR.info(msg)
+
+
+@due.dcite(
+     Doi(''),
+     path='phys2denoise',
+     description='Creation of regressors for physiological denoising',
+     version=__version__,
+     cite_module=True)
+def phys2denoise(filename, metric_args, outdir='.',
+                 metrics=[crf, rpv, rv, rvt, rrf, 'retroicor_card', 'retroicor_resp'],
+                 debug=False, quiet=False):
+    """
+    Run main workflow of phys2denoise.
+
+    Runs the parser, does some checks on input, then computes the required metrics.
+
+    Notes
+    -----
+    The code was greatly copied from phys2bids (copyright the physiopy community)
+
+    """
+    # Check options to make them internally coherent pt. I
+    # #!# This can probably be done while parsing?
+    outdir = os.path.abspath(outdir)
+    os.makedirs(outdir)
+    os.makedirs(os.path.join(outdir, 'code'))
+    log_path = os.path.join(outdir, 'code', 'logs')
+    os.makedirs(log_path)
+
+    # Create logfile name
+    basename = 'phys2denoise_'
+    extension = 'tsv'
+    isotime = datetime.datetime.now().strftime('%Y-%m-%dT%H%M%S')
+    logname = os.path.join(log_path, (basename + isotime + '.' + extension))
+
+    # Set logging format
+    log_formatter = logging.Formatter(
+        '%(asctime)s\t%(name)-12s\t%(levelname)-8s\t%(message)s',
+        datefmt='%Y-%m-%dT%H:%M:%S')
+
+    # Set up logging file and open it for writing
+    log_handler = logging.FileHandler(logname)
+    log_handler.setFormatter(log_formatter)
+    sh = logging.StreamHandler()
+
+    if quiet:
+        logging.basicConfig(level=logging.WARNING,
+                            handlers=[log_handler, sh],
+                            format='%(levelname)-10s %(message)s')
+    elif debug:
+        logging.basicConfig(level=logging.DEBUG,
+                            handlers=[log_handler, sh],
+                            format='%(levelname)-10s %(message)s')
+    else:
+        logging.basicConfig(level=logging.INFO,
+                            handlers=[log_handler, sh],
+                            format='%(levelname)-10s %(message)s')
+
+    version_number = _version.get_versions()['version']
+    LGR.info(f'Currently running phys2denoise version {version_number}')
+    LGR.info(f'Input file is {filename}')
+
+    # Save call.sh
+    arg_str = ' '.join(sys.argv[1:])
+    call_str = f'phys2denoise {arg_str}'
+    f = open(os.path.join(log_path, 'call.sh'), "a")
+    f.write(f'#!bin/bash \n{call_str}')
+    f.close()
```

Comment:
Does this work within the main function instead of when the actual command-line arguments are parsed?

Reply:
Yes, that I'm sure about! At least, it does in phys2bids.

Reply:
Even if so, does it make sense to have it? Why would you have a call script if you're running a Python function?

Reply:
From what I recall, this was added during the OHBM hackathon because it is necessary to have the call.sh to be BIDS compatible.

Reply:
Ok, maybe I didn't get it right. @tsalo the idea is that this workflow can be run as a bash script (if you want to).
But I see your point: if you're using python and python only, you probably don't need to get this call. So we should move it to the parser instead. Did I get it right?
If so, once the parser is merged, we'll move it. And we can do the same in `phys2bids`

Reply:
Sorry for the delay in responding- I somehow missed the last few comments here. @smoia I think that's the way to go- put the call.sh saving step in the parser instead of in the Python function.